### PR TITLE
Swift 4.2

### PR DESCRIPTION
**Pull request by maurovc**

### CODE REVIEW

File: Pod/Classes/Banner.swift [hunk 1]
```diff
@@ -114,18 +128,32 @@ open class Banner: UIView {
     }
     
     /// The label that displays the banner's title.
-    @objc open let titleLabel: UILabel = {
+    @objc public let titleLabel: UILabel = {
         let label = UILabel()
-        label.font = UIFont.preferredFont(forTextStyle: UIFontTextStyle.headline)
+      
+        #if swift(>=4.2)
+        let textStyle = UIFont.TextStyle.headline
```
Comment (harlanhaskins) [inline Pod/Classes/Banner.swift]: This one can just be changed to ` UIFont.preferredFont(forTextStyle: .headline)` and be backwards-compatible.

File: Pod/Classes/Banner.swift [hunk 2]
```diff
@@ -114,18 +128,32 @@ open class Banner: UIView {
...
         label.numberOfLines = 0
         label.translatesAutoresizingMaskIntoConstraints = false
         return label
         }()
     
     /// The label that displays the banner's subtitle.
-    @objc open let detailLabel: UILabel = {
+    @objc public let detailLabel: UILabel = {
         let label = UILabel()
-        label.font = UIFont.preferredFont(forTextStyle: UIFontTextStyle.subheadline)
```
Comment (harlanhaskins) [inline Pod/Classes/Banner.swift:128]: Ditto here.

File: Pod/Classes/Banner.swift [hunk 3]
```diff
@@ -277,7 +305,14 @@ open class Banner: UIView {
...
             let constraintFormat = "H:|[label]-(8)-|"
-            contentView.addConstraints(NSLayoutConstraint.defaultConstraintsWithVisualFormat(constraintFormat, options: NSLayoutFormatOptions(), metrics: nil, views: ["label": view]))
+          
+            #if swift(>=4.2)
+            let options = NSLayoutConstraint.FormatOptions()
+            #else
+            let options = NSLayoutFormatOptions()
+            #endif
+          
+            contentView.addConstraints(NSLayoutConstraint.defaultConstraintsWithVisualFormat(constraintFormat, options: options, metrics: nil, views: ["label": view]))
```
Comment (harlanhaskins) [inline Pod/Classes/Banner.swift]: I think you can change this to `[]` and still be backwards compatible.

File: Pod/Classes/Banner.swift [hunk 4]
```diff
@@ -369,11 +404,25 @@ open class Banner: UIView {
 }
 
 extension NSLayoutConstraint {
+  #if swift(>=4.2)
+  @objc class func defaultConstraintsWithVisualFormat(_ format: String, options: NSLayoutConstraint.FormatOptions = NSLayoutConstraint.FormatOptions(), metrics: [String: AnyObject]? = nil, views: [String: AnyObject] = [:]) -> [NSLayoutConstraint] {
```
Comment (harlanhaskins) [inline Pod/Classes/Banner.swift]: Same here

**Comment by harlanhaskins** [comment]
Thank you for this!

File: Pod/Classes/Banner.swift (hunk 1 above)
Comment (maurovc) [reply Pod/Classes/Banner.swift]: True

File: Pod/Classes/Banner.swift (hunk 2 above)
Comment (maurovc) [reply Pod/Classes/Banner.swift:128]: +1

File: Pod/Classes/Banner.swift (hunk 4 above)
Comment (maurovc) [reply Pod/Classes/Banner.swift]: +1

**Comment by maurovc** [comment]
Hey! @harlanhaskins I've updated the PR with some of your comments and moved the project again to Swift 4.0 so you're able to merge the code to the main branch.

**Comment by nmanoogian** [comment]
Thank you for your contribution, @maurovc! Several people submitted PRs for this migration and they were all very similar. Version 0.8.2 is now available with these changes.

**Pull request closed without merging**